### PR TITLE
Asset Browser: Extract from Specific Archive and Load Order

### PR DESCRIPTION
**Pull request by dragonzkiller**
### Please check the order. I swear no one has a real clue what the actual order is and we always get different answers. Maybe @psiberx knows 😅 

Updates the asset browser to support pulling files from specific archives (including mod archives). Also will sort the mod browser by the (a?) load order to show the first loaded files up top. This also includes support for `modlist.txt` in `archive/pc/mod` and REDmod mods. No support for REDmod `modlist.txt` as of this update. See example image.

image

### CODE REVIEW

File: WolvenKit.Modkit/Managers/ArchiveManager.cs [hunk 1]
```diff
@@ -156,9 +158,14 @@ public override void LoadFromFolder(DirectoryInfo archiveDir)
                 return;
             }
 
+            var redModFolder = archiveDir.Name.Contains(Path.DirectorySeparatorChar + "mods" + Path.DirectorySeparatorChar);
+
             IsManagerLoading = true;
 
-            var archiveFiles = Directory.GetFiles(archiveDir.FullName, "*.archive").ToList();
+            // TODO: this will still load things in sub-directories beyond "archive" for REDMod mods
```
Comment (effs) [inline WolvenKit.Modkit/Managers/ArchiveManager.cs:165]: Does it matter if we grab non-enabled archives? It'd save users the trouble of having to enable things just to mod them unless it throws something else off.

**Review by effs — COMMENTED** [review]
Couple Q's but looks good! I'm not up on the latest discussion on what the order should be so I can't speak to that :)

File: WolvenKit.Modkit/Managers/ArchiveManager.cs [hunk 2]
```diff
@@ -355,37 +362,104 @@ public override void LoadModsArchives(FileInfo executable, bool analyzeFiles = t
...
                 }
             }
 
-            files.Sort(string.CompareOrdinal);
-            files.Reverse();
+            legacyFiles.Sort(string.CompareOrdinal);
+            legacyFiles.Reverse();
 
-            foreach (var file in files)
+            // load REDmod mods first
```
[the remark this answers is not in the document]
Comment (dragonzkiller) [reply WolvenKit.Modkit/Managers/ArchiveManager.cs:449]: This is only for displaying withing the app itself. Lower level code *should* be handling the actual extraction, but I don't think it quite takes into account `modlist.txt`, for example.

Effectively this sets what the AB will show with the highest priority mod archives on top

File: WolvenKit/Converters/IGameArchiveToMenuItemStringConverter.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+using WolvenKit.Core.Interfaces;
+
+namespace WolvenKit.Converters;
+public class IGameArchiveToMenuItemStringConverter : IValueConverter
+{
+    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        if (value is IGameArchive archive)
+        {
+            if (!archive.Name.Contains('_'))
```
[the remark this answers is not in the document]
Comment (dragonzkiller) [reply WolvenKit/Converters/IGameArchiveToMenuItemStringConverter.cs:15]: We could, yeah. It's a theoretical (very mild) performance tradeoff between checking if there's an underscore and then replacing it, and just trying to replace everything. We're not trying to exactly be bleeding edge here, so either method would be effectively the same

File: WolvenKit.Modkit/Managers/ArchiveManager.cs (hunk 1 above)
Comment (dragonzkiller) [reply WolvenKit.Modkit/Managers/ArchiveManager.cs:165]: For me, I would say yes, it matters. I'm not the only one who sometimes throws old archives for some appearance mods into subfolders to quickly disable them for determining which I like more. Disabling the subfolders prevents random ones the game isn't even loading from even being shown in the AB.

For example, imagine the mess if someone runs the export script and it starts grabbing files that the game isn't even using and shoving it in the material repo. Or setting up an NPV and the same happens. And so on

File: WolvenKit.Modkit/Managers/ArchiveManager.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (effs) [reply WolvenKit.Modkit/Managers/ArchiveManager.cs:449]: Here I'm not sure if this order should be read top-down or not; if it's possible to do something like [psi's pseudocode](https://discord.com/channels/717692382849663036/717720094196760760/1176737831125786635) example structure with an even more spelled-out name like `loadOrderWithPriorityTopDown`, then this section will hopefully document itself.

[the remark this answers is not in the document]
Comment (dragonzkiller) [reply WolvenKit.Modkit/Managers/ArchiveManager.cs:449]: yeah i'll have to fix this. really it's legacy mods "load" first in AB (meaning they go on top). so i have to update that now.

**Review by rfuzzo — APPROVED** [review]
👍

**Pull request merged**